### ISSUE TITLE
Tool Support Targeting Ollama 0.3

### ISSUE THREAD

**Issue by jmont-dev** [jmont-dev/ollama-hpp#2]
Ollama 0.3 has just been released with tool support. This will be added to Ollama.hpp for the next release.

**Comment by JG-Adams** [comment]
Erm, is there a mistake? Did you already implement tool call?

**Comment by jmont-dev** [comment]
Yes, I made a few other PRs for smaller features while I was working on this branch and the bot closed the issue when they were merged in. Working on this for the next round of features after the current bugs are addressed.

**Comment by JG-Adams** [comment]
How is it going? I would like to try this someday. Is it already doable with the manual method?

**Comment by jmont-dev** [comment]
Working on adding both this as well as structured responses from the base API. The two are similar in their implementation. I'm going to merge some changes for stopping streaming gracefully and then will move onto these features next.

**Comment by jmont-dev** [comment]
I have an example of using this now in the `tool-calling` branch. See `examples/function_calling.cpp`. Working on some classes to simplify the function definitions, but for now you can encode function definitions in json under `['tools']` when making a request.

**Comment by JG-Adams** [comment]
Can you also provide instruction for how to acquire these tools? I tried your example and nothing appeared in the assistant' response.

Also, I ran the program and forgot to set the model to use what I currently got and it would automatically pull in a new model. Trouble is, I happened to have had very little space at the moment. So it's probably best if you make it ask the user if they wanted to pull in the model. It would appeared to have occupied some space and I'm not sure where it's at on my linux system.
